### PR TITLE
fix: Add mac ID or similar stuff to track the download in each machine

### PR DESCRIPTION
**Pull request by saikiranakula-amzn**
*Issue #39 

*Description of changes:*
Add new property "customerId" to Metrics to populate unique Id (Mac id) to identify customers

*Testing done:*
Yes
#### Tests
N/A

### CODE REVIEW

File: packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs [hunk 1]
```diff
@@ -12,5 +14,25 @@ public class MetricsBase
...
+        public string CustomerID
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(_customerID))
+                {
+                    _customerID = NetworkInterface.GetAllNetworkInterfaces()
+                                         .Where(nic => nic.OperationalStatus == OperationalStatus.Up && nic.NetworkInterfaceType != NetworkInterfaceType.Loopback)
+                                         .Select(nic => nic.GetPhysicalAddress().ToString())
+                                         .FirstOrDefault();
```
Comment (marknfawaz) [inline packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs:32]: We probably need to generate our own Id instead of this

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs (hunk 1 above)
Comment (huawenm) [reply packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs:32]: Maybe hash of the physical address?

Comment (saikiranakula-amzn) [reply packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs:32]: we were discussing to generate a guid and store it in the config.json and use it to identity the customer

File: packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs [hunk 2]
```diff
@@ -19,20 +20,20 @@ public class MetricsBase
...
             {
-                if (string.IsNullOrWhiteSpace(_customerID))
+                if (string.IsNullOrWhiteSpace(_uniqueID))
                 {
-                    _customerID = NetworkInterface.GetAllNetworkInterfaces()
+                     _uniqueID = Crypto.SHA256(NetworkInterface.GetAllNetworkInterfaces()
                                          .Where(nic => nic.OperationalStatus == OperationalStatus.Up && nic.NetworkInterfaceType != NetworkInterfaceType.Loopback)
                                          .Select(nic => nic.GetPhysicalAddress().ToString())
-                                         .FirstOrDefault();
+                                         .FirstOrDefault());
```
Comment (marknfawaz) [inline packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Models/MetricsBase.cs]: Would this generate the same result on a machine with multiple network interfaces?

**Review by marknfawaz — CHANGES_REQUESTED** [review]

**Review by flywanli — APPROVED** [review]
LGTM

**Review by ShanshanHe — APPROVED** [review]
Thank you very much for spending the time investigating the scenario when both ethernet and wifi NICs are present and making the code deterministic by prioritizing wifi first.

**Review by marknfawaz — APPROVED** [review]

**Review by huawenm — APPROVED** [review]

**Pull request merged**